### DISCUSSION TITLE
PR merge strategy

### DISCUSSION THREAD

**Discussion by plozanol** [github-stats-22/github-stats#31] [Polls]
# Context
when you approve a PR you can choose multiple ways of merging the code. Each has their pros and cons.

## Merge
it will create a new commit with the title "merge: ...", and a loop of the branch commits will be maintained

## Squash
all PR commits will be transposed on the top of the main branch, similar to a cherry-pick of all commits. Also, PR's commits timestamp will be overridden to the actual time.

## rebase
flatten all PR commits into one, and squash it to the main branch. You lose the granularity of each PR change in favor of a more clean main history line with on commit per PR.

**Comment by mikededo** [comment]
Any but rebasing.
